.ado/templates/setup-repo.yml

```diff
@@ -13,3 +13,5 @@ steps:
   - script: |
       yarn
     displayName: 'yarn install'
+
+  - template: .ado/templates/apple-ensure-valid-cocoapods.yml@self
```

Comment:
I think this will get called for all jobs, not just the apple ones. Can you move it to run in just the apple jobs?